lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1034,7 +1035,16 @@ class MongoClientInterface {
 
     internalListObject(bucketName, params, extension, log, cb) {
         const c = this.getCollection(bucketName);
-        const stream = new MongoReadStream(c, params, params.mongifiedSearch);
+        const _params = Array.isArray(params) ? params : [params];
+        let stream;
+        if (_params.length === 1) {
+            stream = new MongoReadStream(c, _params[0], _params.mongifiedSearch);
+        } else {
+            const masterStream = new MongoReadStream(c, _params[0], _params.mongifiedSearch);
+            const versionStream = new MongoReadStream(c, _params[1], _params.mongifiedSearch);
+            stream = new MergeStream(
```

Comment:
In the `listObject` function, we generate the internal params of the listing algorithm here:
https://github.com/scality/Arsenal/blob/ad62d8b2e0fe9b4f1b4a5536fffa075026c83026/lib/storage/metadata/mongoclient/MongoClientInterface.js#L1131-L1132
Those params depend on the version of the algorithm (DelimiterMaster, DelimiterVersions) as well as the vFormat. In `v0` the params are in form of an object and in `v1` they are in form of an array of two objects where each object contain the filters to apply to the master and version stream.

This is the way it is implemented in the `metadata` backend, and to avoid changing everything i reused the code and classes already there (MergeStream)

Reply:
ok, let's leave it like this then...